elements/lisk-transactions/test/15_pom_transaction.spec.ts

```diff
@@ -12,59 +12,494 @@
+			expect(status).toBe(Status.FAIL);
+			expect(errors).toHaveLength(3);
+			expect(errors[2].message).toInclude(
+				'Invalid block signature for header 2',
+			);
+		});
+
+		it('should return errors if misbehaving account is not a delegate', async () => {
+			store = new StateStoreMock([sender, { ...delegate, isDelegate: 0 }], {
+				lastBlockHeader: {
+					height:
+						validProofOfMisbehaviorTransactionScenario1.testCases.output.asset
+							.header1.height + 10,
+				} as any,
+				lastBlockReward: BigInt(1),
+			});
 
-		it.todo(
-			'should append height h to pomHeights property of misbehaving account',
-		);
+			const { errors, status } = await transactionWithScenario1.apply(store);
+			expect(status).toBe(Status.FAIL);
+			expect(errors).toHaveLength(3);
+			expect(errors[2].message).toInclude('Account is not a delegate');
+		});
 
-		it.todo('should set isBanned property to true is pomHeights.length === 5');
+		it('should return errors if misbehaving account is already banned', async () => {
+			store = new StateStoreMock(
+				[
+					sender,
+					{ ...delegate, delegate: { ...delegate.delegate, isBanned: true } },
+				],
+				{
+					lastBlockHeader: {
+						height:
+							validProofOfMisbehaviorTransactionScenario1.testCases.output.asset
+								.header1.height + 10,
+					} as any,
+					lastBlockReward: BigInt(1),
+				},
+			);
 
-		it.todo('should return errors if misbehaving account is not a delegate');
+			const { errors, status } = await transactionWithScenario1.apply(store);
+			expect(status).toBe(Status.FAIL);
+			expect(errors).toHaveLength(3);
+			expect(errors[2].message).toInclude(
+				'Cannot apply proof-of-misbehavior. Delegate is banned.',
+			);
+		});
 
-		it.todo('should return errors if misbehaving account is already banned');
+		it('should return errors if misbehaving account is already punished at height h', async () => {
+			store = new StateStoreMock(
+				[
+					sender,
+					{
+						...delegate,
+						delegate: {
+							...delegate.delegate,
+							pomHeights: [
+								validProofOfMisbehaviorTransactionScenario1.testCases.output
+									.asset.header1.height + 10,
+							],
+						},
+					},
+				],
+				{
+					lastBlockHeader: {
+						height:
+							validProofOfMisbehaviorTransactionScenario1.testCases.output.asset
+								.header1.height + 10,
+					} as any,
+					lastBlockReward: BigInt(1),
+				},
+			);
 
-		it.todo(
-			'should return errors if misbehaving account is already punished at height h',
-		);
+			const { errors, status } = await transactionWithScenario1.apply(store);
+			expect(status).toBe(Status.FAIL);
+			expect(errors).toHaveLength(3);
+			expect(errors[2].message).toInclude(
+				'Cannot apply proof-of-misbehavior. Delegate is already punished.',
+			);
+		});
 
-		it.todo('should return errors if |header1.height - h| >= 260000');
+		it('should add reward to balance of the sender', async () => {
```

Comment:
there should be also test to check correct signature and id against the protocol spec

Reply:
do you mean block signature? that is already checked. the tx signature is checked in base tx tests.

Reply:
I mean the transaction signature.
Because getAssetBytes changes the signature, complete signature should be checked against the protocol spec in each transaction too